src/kademlia/dht_tracker.cpp

```diff
@@ -117,14 +117,16 @@ namespace libtorrent { namespace dht {
 	{
 		address local_address = s->get_local_endpoint().address();
 #if TORRENT_USE_IPV6
-		// don't try to start dht nodes on non-global addresses
+		// don't try to start dht nodes on non-global IPv6 addresses
+		// with IPv4 the interface might be behind NAT so we can't skip them based on the scope of the local address
+		// and we might not have the external address yet
```

Comment:
makes sense. I just seem to recall some default behavior to enumerate all interfaces and create sockets on them, but maybe that's gone now.